proposals/lip-introduce_the_auth_module.md

```diff
@@ -0,0 +1,235 @@
+
+
+### Commands
+
+#### registerMultisignatureGroup
+
+This command is specified in [LIP 0017](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0017.md#multisignature-account-registration). It allows users to register a multisignature group for the sender account.
+
+
+##### Parameters
+
+The properties and schema for this command are defined in [LIP 0028](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0028.md#multisignature-registration).
+
+
+##### Verification
+
+The verification of this command is defined in [LIP 0017](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0017.md#multisignature-account-registration).
+
+
+##### Execution
+
+This command updates the entries in the sender `authAccount` to the values specified in the command parameters.
+
+### Protocol Logic for Other Modules
+
+#### getAuthAccount
+
+This function is used to retrieve information about an auth account.
+
+##### Parameters
+
+* `address`: A 20 bytes value identifying the auth account.
+
+
+##### Returns
+
+This function returns `authAccount(address)`. If there is no entry corresponding to `address`, it returns an empty object.
+
+### Endpoints for Off-Chain Services
+
+#### getAuthAccount
+
+This function is used to retrieve information about an auth account.
+
+##### Parameters
+
+* `address`: A 20 bytes value identifying the auth account.
+
+##### Returns
+
+This function returns `authAccount(address)`. If there is no auth account corresponding to `address`, it returns an empty object.
+
+#### verifySignatures
+
+This function verifies the signatures of a given transaction `trs`, including transactions from  multisignature accounts.
+
+##### Parameters
+
+* `trs`: A transaction object.
+
+##### Returns
+
+This function returns `true` if the transaction object contains a valid signature, `false` otherwise, as specified in [LIP 0017](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0017.md#transaction-creation-and-verification).
+
+
+#### verifyNonce
+
+This function verifies the nonce of a given transaction `trs` and returns a boolean value.
+
+
+##### Parameters
+
+* `trs`: A transaction object.
+
+##### Returns
+
+This function returns `true` if the transaction object contains a valid nonce, `false` otherwise. This is done using the `isValidNonce` function defined [below](#transaction-verification).
+
+
+### Block Processing
+
+#### Transaction Verification
+
+As part of the verification of a transaction `trs`, the following checks are applied. If the checks fail the transaction is invalid and the whole block is discarded. 
+
+* Check that the transaction nonce is equal to the account nonce by returning `isValidNonce(trs)`.
+
+```python
+isValidNonce(trs):
+    let senderAddress be the address corresponding to trs.senderPublicKey
+    if there is no entry in the in the auth data substore with storeKey == senderAddress:
+        return True
```

Comment:
I think this still should check that `trs.nonce == 0`.

Reply:
it is true it is not a security issue not to enforce it, but I think it makes sense

Reply:
thanks!